src/blueapi/log.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import os
+
+from graypy import GELFTCPHandler
+
+from blueapi.config import LoggingConfig
+
+
+class BeamlineTagFilter(logging.Filter):
+    """Filter to attach beamline attribute to LogRecords.
+
+    Attaches attribute `beamline` to all LogRecords that are passed through. Value is
+    taken from `BEAMLINE` env var, defaulting to `dev`.
+    """
+
+    beamline: str | None = os.environ.get("BEAMLINE")
```

Comment:
We need to support BEAMLINE or INSTRUMENT and should be trying to standardise on instrument.